File: backend/src/main/java/vn/kurisu/mentormatch/service/DisputeService.java
package vn.kurisu.mentormatch.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import vn.kurisu.mentormatch.dto.request.CreateDisputeRequest;
import vn.kurisu.mentormatch.dto.request.CounterDisputeRequest;
import vn.kurisu.mentormatch.dto.request.ResolveDisputeRequest;
import vn.kurisu.mentormatch.dto.response.ApiResponse;
import vn.kurisu.mentormatch.dto.response.DisputeResponse;

public interface DisputeService {
    ApiResponse<DisputeResponse> createDispute(CreateDisputeRequest request);
    ApiResponse<DisputeResponse> counterDispute(Integer disputeId, CounterDisputeRequest request);
    ApiResponse<DisputeResponse> getDisputeByBooking(Integer bookingId);
    ApiResponse<Page<DisputeResponse>> getMyDisputes(Pageable pageable);
    ApiResponse<Page<DisputeResponse>> getMentorDisputes(Pageable pageable);
    
    // Admin functions
    ApiResponse<Page<DisputeResponse>> getAllDisputes(Pageable pageable);
    ApiResponse<DisputeResponse> resolveDispute(Integer disputeId, ResolveDisputeRequest request);
}

File: frontend/src/services/disputeService.js
import api from "./api";

export const disputeService = {
  createDispute: async (data) => {
    // data: { bookingId, reason }
    const response = await api.post("/disputes", data);
    return response;
  },

  getMyDisputes: async (params) => {
    const response = await api.get("/disputes/my-disputes", { params });
    return response;
  },

  getAllDisputes: async (params) => {
    // Admin only
    const response = await api.get("/admin/disputes", { params });
    return response;
  },

  getMentorDisputes: async (params) => {
    const response = await api.get("/disputes/mentor-disputes", { params });
    return response;
  },

  getDisputeByBooking: async (bookingId) => {
    const response = await api.get(`/disputes/booking/${bookingId}`);
    return response;
  },

  resolveDispute: async (disputeId, data) => {
    // Admin only
    // data: { adminNote, acceptRefund: true/false }
    const response = await api.post(
      `/admin/disputes/${disputeId}/resolve`,
      data,
    );
    return response;
  },

  counterDispute: async (disputeId, data) => {
    // data: { counterReason }
    const response = await api.post(`/disputes/${disputeId}/counter`, data);
    return response;
  },
};
